src/elements/content-sidebar/activity-feed/activity-feed/AnnotationActivityLinkProvider.js

```diff
@@ -0,0 +1,43 @@
+// @flow
+import * as React from 'react';
+import AnnotationActivityLink from '../annotations/AnnotationActivityLink';
+
+import annotationsMessages from '../annotations/messages';
+
+import type { Annotation } from '../../../../common/types/feed';
+
+type AnnotationActivityLinkProviderProps = {
+    isCurrentVersion: boolean,
+    item: Annotation,
+    onSelect: (annotation: Annotation) => void,
+};
+
+const AnnotationActivityLinkProvider = ({ isCurrentVersion, item, onSelect }: AnnotationActivityLinkProviderProps) => {
+    const { file_version, id, target } = item;
+
+    const isFileVersionUnavailable = file_version === null;
+
+    const linkMessage = isCurrentVersion
+        ? annotationsMessages.annotationActivityPageItem
+        : annotationsMessages.annotationActivityVersionLink;
+    const linkValue = isCurrentVersion ? target?.location.value : file_version?.version_number;
```

Comment:
does location need an optional chain?

Reply:
No, it doesn't. Target is a union of type:
```
export type Target = TargetRegion | TargetPoint;
```
Both of those types require location:
```
export type TargetRegion = {
    location: Page,
    shape?: Rect,
    type: 'region',
};

export type TargetPoint = {
    location: Page,
    type: 'point',
    x: number,
    y: number,
};
```